### PR TITLE
bugfix: handle new rate-limits for querying pools

### PR DESCRIPTION
**Pull request by akorchyn**
Closes: #431 

* Added fast-near for testnet
* Rate limited querying pools to 140 / 30s

Ideally, if Fastnear is stable enough, the user shouldn't be able to hit this limit as we will query information within the pools the user has staked before.
But in case the fastener fails, we would need to query all active pools, which number more than 150.

| Network | FastNear | Non fast near |
| :---:   | :---: | :---: |
| Testnet | < 140 (mostly 0-5) | 563 pools 
| Mainnet | < 140 ( mostly 0-5 ) | State of contract poolv1.near is too large to be viewed |

@race-of-sloths

### CODE REVIEW

**Comment by race-of-sloths** [comment]
@akorchyn Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/akorchyn?type=bot&pr=near/near-cli-rs/433&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/akorchyn?type=bot&pr=near/near-cli-rs/433&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/akorchyn)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/akorchyn)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: src/commands/account/view_account_summary/mod.rs
Comment (frolvanya) [inline src/commands/account/view_account_summary/mod.rs:174]: Right now if one batch fails, then everything else fails as well. I'm not sure if this is okay, let's at least add a retry logic. Ideally, it'd be great to notify user if one batch failed after retries and move on to the next batch. What do you think about it?

File: src/config/migrations.rs
Comment (frolvanya) [inline src/config/migrations.rs:109]: Not sure if adding a minor/patch version for configs is a good idea. I'd keep only major version incremental like `v3`, `v4` and so on, but this is not very critical, so if you have a strong opinion that it should be `v2.1`, then let's keep it that way

**Review by frolvanya — CHANGES_REQUESTED** [review]

**Comment by frolvanya** [comment]
@race-of-sloths score 3

File: src/config/migrations.rs
Comment (akorchyn) [reply src/config/migrations.rs:109]: We don't have any changes in the fields that's why I decided to go with a minor version. But I don't have strong opinion as well

Comment (frolvanya) [reply src/config/migrations.rs:109]: Also, I think update should be done with `impl From<ConfigV2> for ConfigV2_1`, not with a regular function

File: src/commands/account/view_account_summary/mod.rs
Comment (dj8yfo) [reply src/commands/account/view_account_summary/mod.rs:174]: The intial version is hard to understand without running, but looks like it fails too on any error within a single initial batch.
With batching this simple command will take at least 30 seconds now on mainnet, and retries could make this even longer. Suggest skipping displaying delegated_stake field instead of retrying if there has been at least one error, but displaying other info, that was successfully queried

File: src/commands/account/view_account_summary/mod.rs
Comment (dj8yfo) [inline src/commands/account/view_account_summary/mod.rs:135]: suggest renaming `validators`. @akorchyn  you mentioned it takes (it might take) ~10 batches on mainnet to iterate over. With the named batches size (140) this number is too big to represent validators. If it's not `validators` , it should be renamed accordingly

**Review by frolvanya — APPROVED** [review]
LGTM

**Comment by dj8yfo** [comment]
@race-of-sloths score 13

**Comment by PolyProgrammist** [comment]
LGTM. I don't think there should be a separate ConfigV3 struct as it is the same as V2. Though it's ok

**Pull request merged**
